tools/import-verifier/import-verifier.go

```diff
@@ -89,23 +98,58 @@ func (i *ImportRestriction) forbiddenImportsFor(pkg Package) []string {
 //   - of a package under the rootPackage
 //   - is not of the base import path or a sub-package of it
 //   - is not of an allowed path or a sub-package of one
-func (i *ImportRestriction) isForbidden(imp string) bool {
-	importsBelowRoot := strings.HasPrefix(imp, rootPackage)
-	importsBelowBase := strings.HasPrefix(imp, absolutePackage(i.BaseImportPath))
-	importsAllowed := false
-	for _, allowed := range i.AllowedImports {
-		importsAllowed = importsAllowed || strings.HasPrefix(imp, absolutePackage(allowed))
+func (i *ImportRestriction) isAllowed(packageToCheck string) bool {
+	// if its not under our root, then its a built-in.  Everything else is under
+	// github.com/openshift/origin or github.com/openshift/origin/vendor
+	if !strings.HasPrefix(packageToCheck, rootPackage) {
+		return true
+	}
+	if i.isIncludedInRestrictedPackages(packageToCheck) {
+		return true
 	}
 
-	return importsBelowRoot && !importsBelowBase && !importsAllowed
+	for _, allowedPackage := range i.AllowedImportPackages {
+		if strings.HasPrefix(allowedPackage, "vendor") {
+			allowedPackage = rootPackage + "/" + allowedPackage
+		}
+		if packageToCheck == allowedPackage {
+			return true
```

Comment:
The logic here isn't super clear (and it can't be simpler) so that's why I was using named booleans and delegating logic to keep the level of indentation down. As this is the "meat" of the check IMO we should keep it that way:


```go
func (i *ImportRestriction) isAllowed(packageToCheck string) bool {
    return !i.importsBelowRoot(packageToCheck) || i.isIncludedInRestrictedPackages(packageToCheck) || i.isExplicitlyAllowed(packageToCheck)
}

func (i *ImportRestriction) isExplicitlyAllowed(packageToCheck string) bool {
    return i.importsAllowedPackage(packageToCheck) || i.importsBelowAllowedRoot(packageToCheck)
}
```